app/src/main/java/com/cmput301f20t21/bookfriends/ui/add/AddEditActivity.java

```diff
@@ -0,0 +1,123 @@
+    private Button saveButton;
+    private Button viewRequestButton;
+
+    private ImageView bookImage;
+
+    private TextView titleTextView;
+    private TextView isbnTextView;
+    private TextView authorTextView;
+
+    private TextInputLayout isbnEditText;
+    private TextInputLayout titleEditText;
+    private TextInputLayout authorEditText;
+    private TextInputLayout descriptionEditText;
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.add_edit_activity);
+        bookImage = findViewById(R.id.imageView); // will be replaced with actual image
+        titleTextView = findViewById(R.id.title_text_view);
+        isbnTextView = findViewById(R.id.isbn_text_view);
+        authorTextView = findViewById(R.id.author_text_view);
+        isbnEditText = findViewById(R.id.ISBN_layout);
+        titleEditText = findViewById(R.id.title_layout);
+        authorEditText = findViewById(R.id.author_layout);
+        descriptionEditText = findViewById(R.id.description_layout);
+
+        cancelButton = findViewById(R.id.cancel_button);
+        scanButton = findViewById(R.id.scanner_button);
+        saveButton = findViewById(R.id.save_button);
+        viewRequestButton = findViewById(R.id.view_request_button);
+
+        cancelButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                backToHome();
+            }
+        });
+
+        saveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                saveInformation();
+            }
+        });
+
+        scanButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                openScanner();
+            }
+        });
+
+        viewRequestButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                // TODO: implement view all requests
+                // open new activity
+            }
+        });
+    }
+
+    /**
+     * when cancel button is clicked, go back to the home screen
+     */
+    public void backToHome() {
+        finish();
+    }
+
+    /**
+     * Users should be able to manually fill in or user the scanner to retrieve the information
+     * for the book they want to add
+     * After checking all required fields, save button is clicked
+     * TODO: this is just a placeholder
+     */
+    public void saveInformation() {
+        String isbn = isbnEditText.getEditText().getText().toString();
+        String title = titleEditText.getEditText().getText().toString();
+        String author = authorEditText.getEditText().getText().toString();
+        if (isbn.length() == 0) {
+            isbnEditText.setError("Cannot be empty");
```

Comment:
use string resource here too, anything that need to be displayed should never be hardcoded

Reply:
A reminder to change the strings to string resource :) 

there should already be a row in string.xml that have the text " cannot be empty"

remember to pull master